### DISCUSSION TITLE
mdx supports importing source code of files

### DISCUSSION THREAD

**Discussion by asurance** [web-infra-dev/rspress#1109] [Ideas]
目前mdx中代码块编辑因为没有提示，会导致维护不太方便，如果在文档外用专门的代码文件维护，mdx能展示也是导入文件,看不到对应源码，例如
```tsx
import Demo from 'demos';
export default Demo;
```
后续会考虑支持从代码文件直接引入源码吗

**Comment by Timeless0911** [comment]
We may need to support a syntax to introduce code snippets features, which requires implementing the corresponding syntax parsing in mdx-rs and writing a remark plugin for mdx-js. 

You can create an issue to track on it, or try to contribute a PR.

**Reply by asurance** [reply]
https://github.com/unlight/remark-sources 找到一个这个插件，不过
1. 没有了热重载能力，这个我理解应该是插件问题
2. 他采用的语法是
<pre>
```tsx (...path)
```
</pre>
不确定和你们已有的一些定义冲突
3. 另外，你们如何处理mdx-rs和mdx-js之间的兼容问题，比如这个功能mdx-rs内置支持了，而mdx-js确依赖插件支持。而使用方可能一开始用的默认的mdx-rs，之后由于使用插件而降级到mdx-js，导致这个功能不可用

**Reply by Timeless0911** [reply]
- 目前应该支持的就是代码块高亮和行号显示 https://rspress.dev/zh/guide/basic/use-mdx#%E4%BB%A3%E7%A0%81%E5%9D%97
- 语法上我认为比较理想的应该还是变量声明的方式，这样不容易和其他相关插件起冲突
<pre>
``` path="foo/bar.js"
```
</pre>
- 目前我们提供的功能如果不支持 mdxRs 会自动降级到 mdxjs 上，如果有功能 mdx-rs 支持了，那么这个功能在 mdx-js 的插件应该被内置进 rspress 中

**Reply by asurance** [reply]
我看了下mdx-rs的代码，我看目前似乎内部插件是默认都开启，也无法从外部传入配置是吧

**Reply by Timeless0911** [reply]
是的

**Comment by SoonIter** [comment]
supported in https://github.com/web-infra-dev/rspress/releases/tag/v2.0.0-beta.21

hmr implementation: https://github.com/web-infra-dev/rspress/pull/2423
